engine/app/controllers/good_job/active_jobs_controller.rb

```diff
@@ -4,5 +4,10 @@ def show
       @jobs = GoodJob::Job.where("serialized_params ->> 'job_id' = ?", params[:id])
                           .order(Arel.sql("COALESCE(scheduled_at, created_at) DESC"))
     end
+
+    def destroy
+      GoodJob::Job.where("serialized_params ->> 'job_id' = ?", params[:id]).delete_all
```

Comment:
I used the same lookup as for show page, but why are we using AJ job id here? Wouldn't be easier to just use db id?

Reply:
When an ActiveJob is retried, it generates a new GoodJob resource. That GJ resource has a new primary ID, but the ActiveJob ID is the same. In other words, a single ActiveJob job's history is represented across multiple GoodJob job entries.

I think you have the implementation right here, by deleting all of the GoodJob resources that have the same ActiveJob ID.

...but it's not pretty. And when thinking about "retry" in the future, we have to make sure that it's operating on the latest/newest/head GoodJob resource for a particular ActiveJob job.

To put it another way, `GoodJob::Job` resources are "runs".

Reply:
This also makes me wonder about the safety of having the "delete" on the GJ Job, rather than on the Show page itself to represent that it's deleting _all_ the runs. But this also doesn't have to be perfect.

I think the worry I have is that someone will delete a job that has another run that is currently executing or waiting to execute.

Generative/blue sky idea: delete could be wrapped with an advisory lock on all the matching records.

Reply:
Ah right. We have `GoodJob.preserve_job_records = :on_unhandled_error` so I never seen this.

So I think that for deleting we should use record id to delete only given record, as I believe this is what would be expected, by the principle of least surprise. WDYT?

Reply:
I changed it, to remove only one, given job.
I believe to be more REST-ful we could change the show action to be actually the index one with applied filter.

Reply:
Thanks, I think that's least surprise to only remove the one job 👍 

I want to suggest moving the destroy action into `good_jobs#destroy` because it is acting upon a good_job record. It could be clearer by having a `GoodJobJobsController` and an `ActiveJobJobsController` maybe?

I want to be fairly permissive at this stage because I'm learning both the functionality that people want, and it's also helpful to think through these things with you on actual PRs 😄

Reply:
That makes sense to not mix both, however what do you think about:
>  I believe to be more REST-ful we could change the show action to be actually the index one with applied filter.

This way all the actions fit into one controller.
Also currently the "show" action can display many jobs, which might be confusing.

Reply:
In my mind, the `active_jobs#show` controller is still displaying a single ActiveJob Job, of which the details are multiple GoodJob Jobs.

I am imagining that this page would, in the future, also have some roll-up statistics (when the ActiveJob Job was initially enqueued, whether it has finished running, total number of retries, etc.).

As an end-user, the ActiveJob job is _the thing_ I imagine most users are conceptualizing and the GoodJob Job is the a "run" or "attempt". Unfortunately there isn't a single database record that models the ActiveJob Job yet for the purpose of display.

Reply:
Thanks for explanation. I moved it out to separate controller